libbeat/common/file/interval_rotator.go

```diff
@@ -0,0 +1,209 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package file
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strconv"
+	"time"
+)
+
+type intervalRotator struct {
+	interval    time.Duration
+	lastRotate  time.Time
+	fileFormat  string
+	clock       clock
+	weekly      bool
+	arbitrary   bool
+	newInterval func(lastTime time.Time, currentTime time.Time) bool
+}
+
+type clock interface {
```

Comment:
Why does this interface exists? What's the reason of having a `realClock`? Why not simply call `time.Now` directly instead of through an interface?

Reply:
I assume it's used during unit testing.

Reply:
It's only for testing. I'm happy to do it differently if there's a better or more idiomatic way in Go.